lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -0,0 +1,407 @@
+                            tag_node = NULL;
+                        }
+                    }
+                    else
+                    {
+                        tag_node = NULL;
+                    }
+                }
+                else
+                {
+                    tag_node = NULL;
+                }
+            }
+
+            // Throw exception if node not found
+            if (tag_node == NULL)
+            {
+                // Given file does not exist
+                throw eprosima::qosprof::ElementNotFound(
+                    tag_name + " does not have an element with key " + att_value + "\n");
+
+                return NULL;
+            }
+            else
+            {
+                // Return Node
+                return tag_node;
+            }
+        }
+        // Obtain node based on LIST
+        else
+        {
+            int32_t realIndex = index;
+
+            // Transform index to real index
+            if (index < 0)
+            {
+                realIndex = node_tag_list->getLength() + index;
+            }
+
+            // Check bounds
+            if (realIndex < 0 || realIndex >= node_tag_list->getLength())
+            {
+                // Throw eprosima::qosprof::ElementNotFound exception
+                throw eprosima::qosprof::ElementNotFound(
+                    tag_name + " does not have an element in position " + std::to_string(realIndex) + "\n");
+
+                return NULL;
+            }
+            // Return Node
+            return node_tag_list->item(realIndex);
+        }
+    }
+}
+
+void clear_node(
+        xercesc::DOMNode*& parent_node,
+        xercesc::DOMNode*& node_to_be_deleted)
+
+{
+    // Check if parent node has child nodes
+    if (parent_node->hasChildNodes())
+    {
+        // Check if the node to be deleted is the last child node of the parent
+        if (parent_node->getChildNodes()->getLength() == 1 &&
+            parent_node->getChildNodes()->item(0)->getNodeName() == node_to_be_deleted->getNodeName())
+        {
+            // Invalid request: report exception
+            std::string message = "could not delete last element from ";
+            message.append(xercesc::XMLString::transcode(parent_node->getNodeValue()));
+            message.append(". Delete element by running the predecessor clear API.\n");
+
+            // Throw ElementInvalid exception with custom message
+            throw eprosima::qosprof::ElementInvalid(message);
+            return;
+        }
+    }
+    // Remove node from parent
+    parent_node->removeChild(node_to_be_deleted);
+
+    // Free node resources (optional?)
+    node_to_be_deleted->release();
+    delete node_to_be_deleted;
+}
+
+void reset_node(
+        xercesc::DOMNode*& node)
+
+{
+    // Loop for childs
+    while (node->hasChildNodes())
+    {
+        // Remove all childs
+        node->removeChild(node->getFirstChild());
+    }
+}
+
+xercesc::DOMNode* get_node(
```

Comment:
I suppose this is a question of implementation philosophy. You have opted to have a unique method called for every case and then specific cases that call the generic method. However, maybe another approach, having a specific method for each kind of element (Simple, Complex, List and Maps) can be have because we know what type should be each element. Nevertheless, I am not against keeping the current implementation and discuss later on if we change the approach.

Reply:
Leaving unresolved to keep it in mind.